app/scripts/lib/multichain-api/provider-authorize/handler.js

```diff
@@ -197,7 +202,27 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     });
 
-    // TODO: metrics/tracking after approval
+    // TODO: Contact analytics team for how they would prefer to track this
+    // first time connection to dapp will lead to no log in the permissionHistory
+    // and if user has connected to dapp before, the dapp origin will be included in the permissionHistory state
+    // we will leverage that to identify `is_first_visit` for metrics
+    const isFirstVisit = !Object.keys(
+      hooks.metamaskState.permissionHistory,
+    ).includes(origin);
+    if (shouldEmitDappViewedEvent(hooks.metamaskState.metaMetricsId)) {
+      hooks.sendMetrics({
+        event: MetaMetricsEventName.DappViewed,
+        category: MetaMetricsEventCategory.InpageProvider,
```

Comment:
Also probably want a different category here. TODO: discuss this with analytics team